icu4c/source/common/rbbi_cache.cpp

```diff
@@ -420,50 +427,133 @@ UBool RuleBasedBreakIterator::BreakCache::populateNear(int32_t position, UErrorC
     return true;
 }
 
+static inline bool isNumber(UChar32 ch) {
+    static UChar32 range[2][2] = {
+        {0x0030, 0x0039},    //half width number 0~9
+        {0xFF10, 0xFF19}     //full width number 0~9
+    };
+    for(int32_t i = 0 ; i < 2 ; i++) {
+        if(ch >= range[i][0] && ch <= range[i][1])
+            return true;
+    }
+    return false;
+}
+
+UBool RuleBasedBreakIterator::BreakCache::isCjk(UChar32 ch) {
+    return fCjkEngine->handles(ch);
```

Comment:
As I explain before, this approach force RuleBasedBreakIterator::BreakCache in any condiction to call fBI->getLanguageBreakEngine() to construct a CjBreakEngine, even no text are in that range, this is not acceptable because that path will load mega bytes of dictionary into memory while not needed at all. Why don't you just create your own UnicodeSet for that. Just
UnicodeSet cjSet(UnicodeString(u"[[:Han:][:Hiragana:][:Katakana:]\\u30fc\\uff70\\uff9e\\uff9f]"), status);
but keep it a "private static" so you only need to create once.

Reply:
Thanks. Good suggestion.